[Ee Sung] iP

eesung00 opened the pull request:
# DukeYou

> "DukeYou makes my life easier, I can lie down everyday and be a free guy." - [A Broke Man](https://www.youtube.com/watch?v=GtL1huin9EE)

DukeYou is better than Duke and it comes from hell, allow you to store your evil occasions in your life. It's,
- 100% *VIRUS FREE*
- confidential
- SUPER FAST to use ~~so that u won't get caught~~

All you need to do is,
1. send thoughts and pray to satan through here
2. smash your keyboard and mouse.
3. the app will know your evil thoughts 😈 

**HOWEVER** you need to `run` code this to see if you are qualify to use this app.
```java
public class Main {
    public static void main(String[] args) {
        System.out.println("Can I use this app?");
        System.out.println("No, go back and sleep");
    }
}
```

And it is **FREE**! Dun *paiseh* come and take

Features:
- [x] encrypted reminder 
- [x] in-app VPN
- [ ] auto-generated real life **CASH**! (coming in 500 years) 🤑 

<sub>Disclaimer: Most of the information is true with some risks that it is false</sub>

Commit:
Add Gradle support

eesung00 committed:
Completed Level 1

eesung00 committed:
Completed Level 2

eesung00 committed:
Completed level 3

eesung00 committed:
Completed Level 4/ Remove Records

eesung00 committed:
improved code structure (level 4)

eesung00 committed:
Completed Level 5 & Start Handle Error

eesung00 committed:
Completed level 6

eesung00 committed:
Set up input and expected output file

eesung00 committed:
Add ID attribute

eesung00 committed:
Complete branch level 7

eesung00 committed:
Complete level-8 and wait to change Duke.java

eesung00 committed:
Merge branch 'branch-Level-7'

eesung00 committed:
Merge branch 'branch-Level-8'

eesung00 committed:
Fix merge bug and update BotUI

eesung00 committed:
Add more OOP class for command

eesung00 committed:
Create package

eesung00 committed:
fix bug

eesung00 committed:
Add find feature

eesung00 committed:
Merge remote-tracking branch 'originUpstream/add-gradle-support'

eesung00 committed:
set up gradle

eesung00 committed:
Add JUnit tests

eesung00 committed:
Create jar file

eesung00 committed:
Complete A-JavaDoc branch

eesung00 committed:
Complete branch A-CodingStandard

eesung00 committed:
Merge branch 'branch-A-JavaDoc'

eesung00 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/command/AddCommand.java

Resolution: keep CodingStandard

eesung00 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/data/duke.txt
#	src/main/java/duke/storage/FileManager.java
#	text-ui-test/runtest.bat

eesung00 committed:
Update messages for level 9 and fix warning

eesung00 committed:
Add BotUI javaDoc, Change BotUI design, Resolve checkstyle

eesung00 committed:
Add GUI Skeleton

eesung00 committed:
Add basic working GUI and modify BotUI to accommodate the application

eesung00 committed:
Fix checkstyle

eesung00 committed:
Revert "Fix checkstyle"

This reverts commit 323cc24bfe6eaafa3b0058139fedaadefc65c82c.

eesung00 committed:
Revert "Revert "Fix checkstyle""

This reverts commit 4e94b2f6b9f5f17211cd922384b8230492ef1509.

eesung00 committed:
Improve GUI (iteration 1)

eesung00 committed:
Merge branch 'branch-Level-10'

ConradLew commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,86 @@
+                Command c = Parser.parse(ui.readCommand());
+                System.out.println(c.execute(taskList, ui));
+                FileManager.write(this.taskList);
+                exitDuke = c.isExit();
```
I like how the SLAP principle is followed here (and also in many other areas of the codebase). The code is very clear and readable.

ConradLew commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,83 @@
+     *                 to print on the user interface
+     * @return String of suitable response according to the user input through BotUI object.
+     * @throws DukeException - thrown from Parser.extractDateTime methods.
+     * @see Parser - the details of the extratDateTime method throw DukeException
```
Minor nitpick of misspelled method name.

eesung00 replied:
Noted! Thanks for the recommendation!

ConradLew commented on src/main/java/duke/ui/BotUI.java:
```diff
@@ -0,0 +1,162 @@
+     * @param taskList the updated TaskRecords to show user the updated number of tasks in it.
+     * @return The String of task added.
+     */
+    public String successRemoved(TaskRecords taskList, Task task) {
```
Method name can be confusing to readers unfamiliar with the bot. Perhaps a more intuitive name could be used here?

eesung00 replied:
Okay, will try to think more about other code reader perspectives

ConradLew reviewed (commented):
LGTM. Minor issues of variable/method naming and spelling present, but overall code quality is excellent with many good practices.

lulucopter commented on src/main/java/duke/common/InputChecker.java:
```diff
@@ -0,0 +1,43 @@
+
+    private static void checkDeadlineAndEvent(String input, String phrase, String taskSplitter) throws DukeException {
+        String description = input.replace(phrase, "").replace(" ", "");
+        if (!description.contains(taskSplitter) || description.substring(0, description.indexOf(taskSplitter)).isEmpty()
```
Minor nitpick, but you can improve the readability of this by defining self-explanatory booleans to store your conditionals

lulucopter commented on src/main/java/duke/common/Messages.java:
```diff
@@ -0,0 +1,35 @@
+            + "(eg. 2022-08-21 1300)";
+    public static final String MESSAGE_FOUND_TASK = "Here you go! your matching tasks in your list";
+    public static final String MESSAGE_NOT_FOUND = "Nothing in the list match your search keyword!";
+}
```
Thought this was a **great** way to store your standard messages. Nice!

lulucopter commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,117 @@
+        try {
+            String[] commandAndDetail = rawInput.split(" ", 2);
+            String command = commandAndDetail[0];
+            if (commandAndDetail.length < 2) {
```
Felt that you can improve readability on this by converting this to a guard clause

lulucopter commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,117 @@
+                return new FindCommand(command, detail);
+            } else {
+                throw new DukeException(UI.invalidFormat());
+            }
```
Code is well formatted and the abstraction makes it easy to read. Would recommend you to explore using `switch case` syntax as well

lulucopter reviewed (commented):
LGTM! Overall, was a joy reading your code as its very clean and self-explanatory. Liked how comprehensive your JUnit tests were as well! Good job!

eesung00 committed:
Improve code quality

Code quality is not ideal.

* There are several class name affect the readability of code.
(e.g. TaskRecords naming is not intuitive)
* Magic number exists in the class.
* Boolean expression is complicated (InputChecker class).
* Calculation can be extracted to improve readability. (Parser class)
* Improper attribute naming with plural (e.g. Todo: "details").

* Classes name are modified. (Deadline, Event, Todo, TaskList).
* Declared private static attribute for magic number.
* Add variables of boolean with intuitive naming.
* Extracted the calculation to private method.
* Modify the attribute naming to avoid confusion.

Using code quality guidelines improve code readability and lesser
regression error in the future modification.

Refer to https://nus-cs2103-ay2223s1.github.io/website/schedule/week5/
for the code quality guidelines.

eesung00 committed:
Merge pull request #1 from eesung00/branch-A-CodeQuality

Improve code quality

eesung00 committed:
Add assertions

The current code does not utilize assertions

* Adding assertions in default of switch-case statements
* Adding assertions in some methods to ensure assumption is not violated

Let's add the assertions to the class methods to check for potential
runtime error

Using assertions can make the code less error prone as it can notice
potential hidden error.

eesung00 committed:
Merge pull request #2 from eesung00/branch-A-Assertions

Add assertions

eesung00 committed:
Merge branch 'branch-A-Assertions'

eesung00 committed:
Add B-DetectAnomalies extension

Tasks adding into the list are not check whether the current taskList
already consist it or not. Task with same timing will also cause issue
to the user (e.g. two event happening at the same time).

Let's do the following modifications:
* Add AnomaliesManager, CancelCommand and UnknownCommand to handle
anomalies.
* AddCommand execute method will do the checking for anomalies.

User will request confirmation by Duke if two tasks timing is close or
the task with same detail is already exist. This alert user that the
task he/she wanted to add have clash with the existing tasks.

eesung00 committed:
Merge pull request #3 from eesung00/branch-BCD-Extension

Add B-DetectAnomalies extension

eesung00 committed:
Merge branch 'master' of https://github.com/eesung00/ip

eesung00 committed:
Add User Guide

eesung00 committed:
Add tasks' JUnit test

eesung00 committed:
Improve code quality

eesung00 committed:
Improve GUI

Current GUI is simple and does not improve user experiences

Let's
* Add different background color to Duke and User dialogbox.
* Make the scrollpane resizable with its parent.
* Make Duke and User profile picture round.

The aspects improved are:
* Easier to differentiate the conversation.
* Allow user to expand chat history without scrolling if necessary.
* Better appearance of user profile

eesung00 committed:
Resolve DialogBox checkstyle

eesung00 committed:
Merge pull request #5 from eesung00/branch-A-BetterGui

Branch A-better GUI

eesung00 committed:
Merge pull request #6 from eesung00/branch-A-MoreTesting

Add tasks' JUnit test

eesung00 committed:
Merge branch 'master' of https://github.com/eesung00/ip

eesung00 committed:
Merge pull request #7 from eesung00/branch-A-UserGuide

Add User Guide

eesung00 committed:
Merge branch 'master' of https://github.com/eesung00/ip

eesung00 committed:
Fix mistake of removing file during previous merge

eesung00 committed:
Reformat UserGuide to fix rendering issue

eesung00 committed:
Change some syntax to Html

eesung00 committed:
Fix Rendering issue and Add Ui.png

eesung00 committed:
Finalize user guide

eesung00 committed:
finalize code and fix read, write file bug.

eesung00 committed:
Edit user guide

eesung00 committed:
Modify text-ui-test

eesung00 committed:
Improve JavaDoc coding style

eesung00 committed:
Modify JavaDoc
